shelley/chain-and-ledger/executable-spec/src/Shelley/Spec/Ledger/STS/Utxow.hs

```diff
@@ -160,10 +171,11 @@ utxoWitnessed =
         ?! MissingScriptWitnessesUTXOW
 
       -- check VKey witnesses
-      verifiedWits tx ?! InvalidWitnessesUTXOW
+      let failedWits = verifiedWits tx
+      failedWits == mempty ?! InvalidWitnessesUTXOW failedWits
 
       let needed = witsVKeyNeeded utxo tx genDelegs
-      needed `Set.isSubsetOf` witsKeyHashes ?! MissingVKeyWitnessesUTXOW
+      needed `Set.isSubsetOf` witsKeyHashes ?! MissingVKeyWitnessesUTXOW (needed `Set.difference` witsKeyHashes)
```

Comment:
I guess this operation is on small enough sets that we don't care too much about repeating work

Reply:
yea, I thought it read nicer without the let binding, and that the extra work in the error case was fine. but maybe that's silly.